component.js

```diff
@@ -142,6 +142,12 @@ function factory (options) {
     }
 
     var Component = React.createClass(componentObject);
+    
+    // If there is a globally or locally set decorator, apply it to the class
+    var decorator = options.decorate || _decorate;
```

Comment:
It seems `var decorator = options.decorate || _decorate;` is correct, as `options` is the result of `createDefaultArguments` and `decorate` (from `render`) is passed to that function. However, I think you're right, a different name for the render. Also, I think maybe this check should happen in `createDefaultArguments` instead.